test/system/aria/dialog_controller_test.rb

```diff
@@ -0,0 +1,33 @@
+require "application_system_test_case"
+
+class Aria::DialogControllerTest < ApplicationSystemTestCase
+  test "dialog is focus-trapping modal" do
+    visit aria_examples_path
+    sleep 1
```

Comment:
@dhh one thing I _do_ want to flag coming out of this experiment is the race condition between System Tests and the Stimulus autoloader.

In this test's case, the `sleep 1` waits an arbitrary amount of time until the underlying page's `aria--dialog` controller code loads, so that the `click_button("Open Dialog")` line is clicking on a button with a Stimulus controller installed onto it.

I'm struggling to conceptualize a mechanism to synchronize these two events. The button's data- attributes already communicate _which_ controller should be loaded, but does not communicate the state of that loading.

One idea is for the autoloading code to mark elements that have controller identifiers that need to be loaded, and then unmark then when the loading is complete. The flip side of that would be for this gem to override any method in the Capybara that interacts with the page, and prefix it with a call to wait for the page to have 0 marked elements.

That feels invasive, but it's my best idea. Am I missing something more compelling?

Reply:
It turns out this is already a documented issue: https://github.com/hotwired/stimulus-rails/issues/35